### PR TITLE
init: allow update also with .img.gz or .img besides to .tar

### PR DESCRIPTION
**Pull request by vpeter4**
needs double space on storage partition for extracted .img file
KERNEL/SYSTEM files are copied directly from mounted partition to /flash
tested on imx6 project
size of initramfs/kernel increased by ~40kB
if downgrade with image without included .md5 files user needs to create file .nocheck (there is no integrity checks done in this case - can be unsafe)

```
### Checking for updates ###
Found new .tar archive. extracting...
Checking KERNEL.md5...
/storage/.update/KERNEL: OK
Checking SYSTEM.md5...
/storage/.update/SYSTEM: OK
Checking size: OK
### mount filesystem /flash/SYSTEM ... ###
### mount filesystem /storage/.update/SYSTEM ... ###
updating Kernel...
updating System...
### mount filesystem /flash/SYSTEM ... ###
updating Bootloader...
*** updating Device Tree Blob: imx6q-udoo.dtb ...
.................................................
*** updating u-boot image: /flash/u-boot.img ...
*** updating u-boot SPL Blob on: /dev/mmcblk0 ...
35+0 records in
35+0 records out
35840 bytes (35.0KB) copied, 0.050510 seconds, 692.9KB/s
System reboots now...
```
```
### Checking for updates ###
Found new .img.gz archive. extracting...
Mounting system partition...
Checking KERNEL.md5...
/storage/.update/.tmp/mnt/KERNEL: OK
Checking SYSTEM.md5...
/storage/.update/.tmp/mnt/SYSTEM: OK
Checking size: OK
updating Kernel...
updating System...
### mount filesystem /flash/SYSTEM ... ###
updating Bootloader...
*** updating Device Tree Blob: imx6dl-cubox-i.dtb ...
.....................................................
*** updating u-boot image: /flash/u-boot.img ...
*** updating u-boot SPL Blob on: /dev/mmcblk0 ...
35+0 records in
35+0 records out
35840 bytes (35.0KB) copied, 0.050320 seconds, 695.5KB/s
System reboots now...
```
```
### Checking for updates ###
Found new .img file...
Mounting system partition...
Checking KERNEL.md5...
/storage/.update/.tmp/mnt/KERNEL: OK
Checking SYSTEM.md5...
/storage/.update/.tmp/mnt/SYSTEM: OK
Checking size: OK
updating Kernel...
updating System...
### mount filesystem /flash/SYSTEM ... ###
updating Bootloader...
*** updating Device Tree Blob: imx6dl-cubox-i.dtb ...
.....................................................
*** updating u-boot image: /flash/u-boot.img ...
*** updating u-boot SPL Blob on: /dev/mmcblk0 ...
35+0 records in
35+0 records out
35840 bytes (35.0KB) copied, 0.050320 seconds, 695.5KB/s
System reboots now...
```

### CODE REVIEW

**Comment by MilhouseVH** [comment]
I know it's not an issue you've introduced, but at this point:

https://github.com/vpeter4/LibreELEC.tv/blob/img_update1/packages/sysutils/busybox/scripts/init#L674

why do we continue to check free space if `MD5_FAILED` is already `1`?

The test at [line 706](https://github.com/vpeter4/LibreELEC.tv/blob/img_update1/packages/sysutils/busybox/scripts/init#L706) should happen at line 674, and the `check_update()` function should exit immediately if the MD5 has failed.

ie.

````
      ...
        else
          echo "missing ${UPDATE_KERNEL}.md5 or ${UPDATE_SYSTEM}.md5..."
          MD5_FAILED="1"
        fi
      fi

      if [ "$MD5_FAILED" -eq "1" -a "$MD5_NOCHECK" -eq "0" ] ; then
          echo "md5 check failed. normal startup in 30s..."
          do_cleanup
          sync
          usleep 30000000
          return 0
      fi

      # get sizes
      < check sizes, exit immediately if not enough space etc.  >

      # size check is ok
      if [ -b $IMAGE_KERNEL ]; then
        update_partition "Kernel" "$UPDATE_KERNEL" "$IMAGE_KERNEL"
      else
        update_file "Kernel" "$UPDATE_KERNEL" "/flash/$IMAGE_KERNEL"
      fi
      update_file "System" "$UPDATE_SYSTEM" "/flash/$IMAGE_SYSTEM"
      update_bootloader
      do_cleanup
      do_reboot
    fi
  }
```

**Comment by vpeter4** [comment]
Some parts could be done little different and I have them in mind. But at this point I wouldn't touch to much. Just to enable this new functionality.

**Comment by MilhouseVH** [comment]
I think it would be a safe change, to be honest - and since this is going to have to be extensively tested, it saves us doing it again for what would be a minor code re-org. If you're really against it that's fine, but it would make more sense to bail as soon as we've determined the md5 files don't match.

File: packages/sysutils/busybox/scripts/init [hunk 1]
```diff
@@ -549,115 +549,176 @@
...
-        if [ ! -f "$UPDATE_DIR/$UPDATE_KERNEL" -o ! -f "$UPDATE_DIR/$UPDATE_SYSTEM" ] ; then
-          echo "missing ${UPDATE_KERNEL} or ${UPDATE_SYSTEM}... normal startup in 10s"
-          sync
-          usleep 10000000
-          return 0
-        fi
-      fi
+    if ! [ -f "$UPDATE_DIR/$UPDATE_KERNEL" -a -f "$UPDATE_DIR/$UPDATE_SYSTEM" ] &&
+       ! [ -f "$UPDATE_TAR" -o -f "$UPDATE_IMG_GZ" -o -f "$UPDATE_IMG" ]; then
+      return
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/init]: Sorry... `return 0` here. :)

**Comment by lrusak** [comment]
@vpeter4 what state is this in?

**Comment by vpeter4** [comment]
I think it is ready.
Also @MilhouseVH is preparing some extra changes on top of this.

**Comment by lrusak** [comment]
Good do go?

**Pull request merged**